<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interactive multi-channel image visualization in R • cytoviewer</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Interactive multi-channel image visualization in R">
<meta property="og:description" content="cytoviewer">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cytoviewer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/cytoviewer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/BodenmillerGroup/cytoviewer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Interactive multi-channel image visualization in
R</h1>
                        <h4 data-toc-skip class="author">Lasse
Meyer</h4>
            <address class="author_afil">
      Department for Quantitative Biomedicine, University of
ZurichInstitute for Molecular Health Sciences, ETH
Zurich<br><a class="author_email" href="mailto:#"></a><a href="mailto:lasse.meyer@dqbm.uzh.ch" class="email">lasse.meyer@dqbm.uzh.ch</a>
      </address>
                              <h4 data-toc-skip class="author">Nils
Eling</h4>
            <address class="author_afil">
      Department for Quantitative Biomedicine, University of
ZurichInstitute for Molecular Health Sciences, ETH
Zurich<br><a class="author_email" href="mailto:#"></a><a href="mailto:nils.eling@dqbm.uzh.ch" class="email">nils.eling@dqbm.uzh.ch</a>
      </address>
                  
            <h4 data-toc-skip class="date">7 August 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/BodenmillerGroup/cytoviewer/blob/HEAD/vignettes/cytoviewer.Rmd" class="external-link"><code>vignettes/cytoviewer.Rmd</code></a></small>
      <div class="hidden name"><code>cytoviewer.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This R package supports interactive visualization of
      multi-channel images and segmentation masks generated by imaging
      mass cytometry and other highly multiplexed imaging techniques
      using shiny. The cytoviewer interface is divided into image-level
      (Composite and Channels) and cell-level visualization (Masks). It
      allows users to overlay individual images with segmentation masks,
      integrates well with SingleCellExperiment and SpatialExperiment
      objects for metadata visualization and supports image
      downloads.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette introduces the <code>cytoviewer</code> package for
interactive multi-channel image visualization. Images as well as
corresponding segmentation masks generated by imaging mass cytometry
(IMC) and other highly multiplexed imaging techniques can be
interactively visualized and explored.</p>
<p>The <code>cytoviewer</code> package builds on top of the <em><a href="https://bioconductor.org/packages/3.17/cytomapper" class="external-link">cytomapper</a></em>
Bioconductor package and extends the static visualization strategies
provided by <code>cytomapper</code> via an <strong>interactive Shiny
application</strong>. The <code>cytoviewer</code> package leverages the
image handling, analysis and visualization strategies provided by the
<em><a href="https://bioconductor.org/packages/3.17/EBImage" class="external-link">EBImage</a></em>
Bioconductor package and offers interactive image visualization
strategies in a similar fashion as <em><a href="https://bioconductor.org/packages/3.17/iSEE" class="external-link">iSEE</a></em> for
single-cell data. In addition, building up on <em><a href="https://bioconductor.org/packages/3.17/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>,
<em><a href="https://bioconductor.org/packages/3.17/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
and <code><a href="https://rdrr.io/pkg/cytomapper/man/CytoImageList.html" class="external-link">cytomapper::CytoImageList</a></code> classes, the
<code>cytoviewer</code> package integrates into the Bioconductor
framework for single-cell and image analysis.</p>
<div class="section level3">
<h3 id="highly-multiplexed-imaging">Highly multiplexed imaging<a class="anchor" aria-label="anchor" href="#highly-multiplexed-imaging"></a>
</h3>
<p>Highly multiplexed imaging allows simultaneous spatially and
single-cell resolved detection of dozens of biological molecules
(e.g. proteins) in their native tissue context. As a result, these
technologies allow an in-depth analysis of complex systems and diseases
such as the tumor microenvironment <span class="citation">(Jackson et
al. 2020)</span> and type 1 diabetes progression <span class="citation">(Damond et al. 2019)</span>.</p>
<p>Imaging-based spatial proteomics methods <span class="citation">(Moffitt, Lundberg, and Heyn 2022)</span> can be
broadly divided into fluorescent cyclic approaches such as tissue-based
cyclic immunofluorescence (t-CyCIF) <span class="citation">(Lin et al.
2018)</span> and one-step mass-tag based approaches such as multiplexed
ion beam imaging (MIBI) <span class="citation">(Angelo et al.
2014)</span> and IMC <span class="citation">(Giesen et al.
2014)</span>.</p>
<p>Of note, the instructions below will focus on the visualization and
exploration of <strong>IMC data</strong> as an example. However, data
from other technologies such as t-CyCIF or MIBI, which produce
pixel-level intensities and (optionally) segmentation masks, can be
interactively visualized with <code>cytoviewer</code> as long as they
have the appropriate input format (see Section <a href="#DataInputFormat">Data input format</a>).</p>
<div class="section level4">
<h4 id="imaging-mass-cytometry">Imaging mass cytometry<a class="anchor" aria-label="anchor" href="#imaging-mass-cytometry"></a>
</h4>
<p>IMC, an advancement of CyTOF, combines antibodies tagged with
isotopically pure rare earth metals with laser ablation and
mass-spectrometry-based detection to produce high-dimensional images
<span class="citation">(Giesen et al. 2014)</span>. It captures the
spatial expression of over 40 proteins in parallel at a sub-cellular
resolution of 1 μm. Thus, IMC is able to detect cytoplasmic and nuclear
localization of proteins.</p>
</div>
</div>
<div class="section level3">
<h3 id="highly-multiplexed-image-analysis">Highly multiplexed image analysis<a class="anchor" aria-label="anchor" href="#highly-multiplexed-image-analysis"></a>
</h3>
<p>To fully leverage the information contained in IMC and multiplexed
imaging data in general, computational tools are of key importance.</p>
<p>The main analysis steps, irrespective of the biological question,
include 1) Visual inspection of images for quality control, 2) Image
pre-processing and segmentation and 3) Single-cell and spatial analysis
<span class="citation">(Windhager, Bodenmiller, and Eling
2021)</span>.</p>
<p>A comprehensive end-to-end workflow for multiplexed image processing
and analysis with detailed information for every analysis step can be
found <a href="https://bodenmillergroup.github.io/IMCDataAnalysis/" class="external-link">here</a>.</p>
<p>Importantly, the <code>cytoviewer</code> package can support,
simplify and improve any of these analysis steps with its easy-to-use
interactive visualization interface in R.</p>
<p>Below we will showcase an <a href="#ExampleWorkflow">example
workflow</a> that highlights the different functionality and potential
application fields of <code>cytoviewer</code>.</p>
</div>
<div class="section level3">
<h3 id="application-overview">Application overview<a class="anchor" aria-label="anchor" href="#application-overview"></a>
</h3>
<p>The <code>cytoviewer</code> interface is broadly divided into <a href="#ImageLevel">image-level</a> (Composite and Channels) and <a href="#CellLevel">cell-level</a> visualization (Masks). It allows users
to overlay individual images with segmentation masks, integrates well
with <code>SingleCellExperiment</code> and
<code>SpatialExperiment</code> objects for metadata visualization and
supports <a href="#Download">image downloads</a>.</p>
<div class="section level4">
<h4 id="DataInputFormat">Data input format<a class="anchor" aria-label="anchor" href="#DataInputFormat"></a>
</h4>
<p>The <code>cytoviewer</code> package combines objects of <em><a href="https://bioconductor.org/packages/3.17/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>,
<em><a href="https://bioconductor.org/packages/3.17/SpatialExperiment" class="external-link">SpatialExperiment</a></em>
and <code><a href="https://rdrr.io/pkg/cytomapper/man/CytoImageList.html" class="external-link">cytomapper::CytoImageList</a></code> classes (from <em><a href="https://bioconductor.org/packages/3.17/cytomapper" class="external-link">cytomapper</a></em>)
to visualize image- and cell-level information.</p>
<p>The <code>cytoviewer</code> function takes up to <strong>five
arguments</strong>.</p>
<p>Firstly, <code>image</code> refers to a <code>CytoImageList</code>
object containing one or multiple multi-channel images where each
channel represents the pixel-intensities of one marker (proteins in
IMC).</p>
<p>Secondly, <code>mask</code> refers to a <code>CytoImageList</code>
object containing one or multiple segmentation masks. Segmentation masks
are defined as one-channel images containing integer values, which
represent the cell ids or background.</p>
<p>Thirdly, the <code>object</code> entry refers to a
<code>SingleCellExperiment</code> or <code>SpatialExperiment</code>
class object that contains cell-specific metadata in the
<code>colData</code> slot.</p>
<p>Lastly, to match information between the <code>CytoImageList</code>
objects and the
<code>SingleCellExperiment</code>/<code>SpatialExperiment</code> object,
two additional spots can be specified:</p>
<ul>
<li><p><code>img_id</code>: a single character indicating the
<code>colData</code> (of the
<code>SingleCellExperiment</code>/<code>SpatialExperiment</code> object)
and <code>elementMetadata</code> (of the <code>CytoImageList</code>
object) entry that contains the image identifiers. These image ids have
to match between the <code>SingleCellExperiment</code>/
<code>SpatialExperiment</code> object and the <code>CytoImageList</code>
objects.</p></li>
<li><p><code>cell_id</code>: a single character indicating the
<code>colData</code> entry that contains the cell identifiers. These
should be integer values corresponding to pixel-values in the
segmentation masks.</p></li>
</ul>
</div>
<div class="section level4">
<h4 id="DataInputVariation">Data input variations<a class="anchor" aria-label="anchor" href="#DataInputVariation"></a>
</h4>
<p>The functionality of <code>cytoviewer</code> depends on which input
objects are user-provided. Below we describe the <strong>four use
cases</strong> in respect to input objects and functionality.</p>
<p><strong>1. Usage of cytoviewer with images, masks and
object</strong></p>
<p>The <em>full</em> functionality of cytoviewer can be leveraged when
<code>image</code>, <code>mask</code> and <code>object</code> are
provided, which is the main intended use case.</p>
<p>This allows <a href="#ImageLevel">image-level</a> visualization
(Composite and Channels), <a href="#CellLevel">cell-level</a>
visualization, overlaying images with segmentation masks as well as
metadata visualization.</p>
<p><strong>2. Usage of cytoviewer with images only</strong></p>
<p>If only the <code>image</code> object is specified, <a href="#ImageLevel">image-level</a> visualization (Composite and
Channels) is possible.</p>
<p><strong>3. Usage of cytoviewer with images and masks</strong></p>
<p><a href="#ImageLevel">Image-level</a> visualization (Composite and
Channels), overlaying of images with segmentation masks and <a href="#CellLevel">cell-level</a> visualization is feasible when
<code>image</code> and <code>mask</code> objects are provided.</p>
<p><strong>4. Usage of cytoviewer with masks and object</strong></p>
<p>If <code>mask</code> and <code>object</code> are specified, <a href="#CellLevel">cell-level</a> visualization as well as metadata
visualization is possible.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="ExampleWorkflow">Example workflow<a class="anchor" aria-label="anchor" href="#ExampleWorkflow"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>The <code>cytoviewer</code> package can be installed from
<code>Bioconductor</code> via:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"cytoviewer"</span><span class="op">)</span></span></code></pre></div>
<p>The development version of <code>cytoviewer</code> can be installed
from Github via:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"remotes"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"BodenmillerGroup/cytoviewer"</span><span class="op">)</span></span></code></pre></div>
<p>To load the package in your R session, type the following:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BodenmillerGroup/cytoviewer" class="external-link">cytoviewer</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-dataset">Example dataset<a class="anchor" aria-label="anchor" href="#example-dataset"></a>
</h3>
<p>For visualization purposes, we will use a toy dataset provided by the
<em><a href="https://bioconductor.org/packages/3.17/cytomapper" class="external-link">cytomapper</a></em>
package.</p>
<p>The dataset contains 3 images of <span class="math inline">\(100\mu{m}\)</span> x <span class="math inline">\(100\mu{m}\)</span> dimensions with 362 segmented
cells and pixel-intensities for 5 proteins: H3, CD99, PIN, CD8a, and
CDH. It is a small subset from a Type 1 Diabetes dataset <span class="citation">(Damond et al. 2019)</span>.</p>
<p>Pixel-level intensities for all 5 markers (5 channels) are stored in
the <code>pancreasImages</code> object.</p>
<p>The corresponding segmentation masks are stored in the
<code>pancreasMasks</code> object.</p>
<p>All cell-specific metadata are stored in the <code>colData</code>
slot of the corresponding <code>SingleCellExperiment</code> object:
<code>pancreasSCE</code>.</p>
<p>For more detailed information on the dataset, please refer to the
respective documentation (e.g. via <code><a href="https://rdrr.io/pkg/cytomapper/man/pancreasImages.html" class="external-link">?pancreasImages</a></code> or the
vignette of the <em><a href="https://bioconductor.org/packages/3.17/cytomapper" class="external-link">cytomapper</a></em>
package).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example datasets </span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BodenmillerGroup/cytomapper" class="external-link">cytomapper</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreasImages"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreasMasks"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreasSCE"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pancreasImages</span></span></code></pre></div>
<pre><code><span><span class="co">## CytoImageList containing 3 image(s)</span></span>
<span><span class="co">## names(3): E34_imc G01_imc J02_imc </span></span>
<span><span class="co">## Each image contains 5 channel(s)</span></span>
<span><span class="co">## channelNames(5): H3 CD99 PIN CD8a CDH</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pancreasMasks</span> </span></code></pre></div>
<pre><code><span><span class="co">## CytoImageList containing 3 image(s)</span></span>
<span><span class="co">## names(3): E34_mask G01_mask J02_mask </span></span>
<span><span class="co">## Each image contains 1 channel</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pancreasSCE</span></span></code></pre></div>
<pre><code><span><span class="co">## class: SingleCellExperiment </span></span>
<span><span class="co">## dim: 5 362 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(2): counts exprs</span></span>
<span><span class="co">## rownames(5): H3 CD99 PIN CD8a CDH</span></span>
<span><span class="co">## rowData names(4): MetalTag Target clean_Target frame</span></span>
<span><span class="co">## colnames(362): E34_824 E34_835 ... J02_4190 J02_4209</span></span>
<span><span class="co">## colData names(9): ImageName Pos_X ... MaskName Pattern</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="function-call">Function call<a class="anchor" aria-label="anchor" href="#function-call"></a>
</h3>
<p>Here as an example, we call <code>cytoviewer</code> with
<code>image</code>, <code>mask</code> and <code>object</code> data to
leverage all provided functionality.</p>
<p>This setting allows <a href="#ImageLevel">image-level</a>
visualization (Composite and Channels), <a href="#CellLevel">cell-level</a> visualization, overlaying images with
segmentation masks as well as metadata visualization.</p>
<p>For further details, please refer to the <code><a href="../reference/cytoviewer.html">?cytoviewer</a></code>
manual or the <code>Help page</code> within the shiny application.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Use cytoviewer with images, masks and object</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cytoviewer.html">cytoviewer</a></span><span class="op">(</span>image <span class="op">=</span> <span class="va">pancreasImages</span>, </span>
<span>                  mask <span class="op">=</span> <span class="va">pancreasMasks</span>, </span>
<span>                  object <span class="op">=</span> <span class="va">pancreasSCE</span>, </span>
<span>                  img_id <span class="op">=</span> <span class="st">"ImageNb"</span>, </span>
<span>                  cell_id <span class="op">=</span> <span class="st">"CellNb"</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  </span>
<span>  <span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>, launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>  <span class="op">}</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="Interface">Interface<a class="anchor" aria-label="anchor" href="#Interface"></a>
</h3>
<p>The <code>cytoviewer</code> interface is divided into a
<strong>Header</strong>, <strong>Sidebar</strong> and
<strong>Body</strong> section (see Figure below).</p>
<p>The <strong>Header</strong> includes package version information,
access to session information and the help page as well as a
dropdown-menu for <a href="#Download">image downloads</a>.</p>
<p>The <strong>Body</strong> features a Tabset-Panel layout allowing the
user to switch between three image modes: <a href="#ImageLevel">Image-level</a> (Composite and Channels) and <a href="#CellLevel">Cell-level</a> (Mask). Furthermore, the Composite and
Mask tabs have zoom controls.</p>
<p>The <strong>Sidebar</strong> panel is subdivided into four sections:
<em>Sample selection</em>, <em>Image-level</em>, <em>Cell-level</em> and
<a href="#General"><em>General</em></a> controls.</p>
<div class="figure">
<img src="cytoviewer_vignette.png" alt=""><p class="caption">cytoviewer_interface</p>
</div>
</div>
<div class="section level3">
<h3 id="ImageLevel">Image-level visualization<a class="anchor" aria-label="anchor" href="#ImageLevel"></a>
</h3>
<p>Image visualization control is split into <em>basic</em> and
<em>advanced controls</em>.</p>
<p><em>Basic controls</em> supports the selection of up to six
markers/channels for <code>image</code> display. Each marker has color
control settings that allow the user to set contrast, brightness, gamma
and select a channel color.</p>
<p>In the <em>advanced controls</em> part, the user can choose to
overlay the displayed images with provided segmentation
<code>masks</code>. Outline color and mask thickness can be adjusted by
the user. Moreover, the masks can be outlined by cell-specific metadata
provided in <code>colData</code> slot of the <code>object</code>.</p>
<p>Of note, for categorical and continuous metadata entries the user can
choose between discrete colors and continuous color palettes (viridis,
inferno, plasma), respectively.</p>
</div>
<div class="section level3">
<h3 id="CellLevel">Cell-level visualization<a class="anchor" aria-label="anchor" href="#CellLevel"></a>
</h3>
<p>Cell visualization has <em>basic controls</em>.</p>
<p>Here, the user can choose to display the provided segmentation
<code>masks</code>. If an <code>object</code> is provided, the masks can
be colored by cell-specific metadata.</p>
<p>Please note again that for categorical and continuous metadata
entries the user can choose between discrete colors and continuous color
palettes (viridis, inferno, plasma), respectively.</p>
</div>
<div class="section level3">
<h3 id="General">General controls<a class="anchor" aria-label="anchor" href="#General"></a>
</h3>
<p>General controls is subdivided into an <em>Image appearance</em> and
<em>Image filters</em> part.</p>
<p>In the <em>Image appearance</em> section, the user can adjust the
scale bar length and include legend/image titles, while the <em>Image
filters</em> section allows to control pixel-wise interpolation
(default) and apply a Gaussian filter.</p>
</div>
<div class="section level3">
<h3 id="Download">Image download<a class="anchor" aria-label="anchor" href="#Download"></a>
</h3>
<p>The <code>cytoviewer</code> package supports fast and uncomplicated
image downloads.</p>
<p>Download controls are part of the <strong>Header</strong> (see
Section <a href="#Interface">Interface</a>).</p>
<p>The user can specify a file name, select the image of interest
(Composite, Channels, Mask) and the file format (pdf, png). Upon
clicking the download button, a pop-window should appear where the user
can specify the download location.</p>
</div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-apple-darwin20 (64-bit)</span></span>
<span><span class="co">## Running under: macOS Monterey 12.6.7</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: internal</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] cytomapper_1.12.0           SingleCellExperiment_1.22.0</span></span>
<span><span class="co">##  [3] SummarizedExperiment_1.30.2 Biobase_2.60.0             </span></span>
<span><span class="co">##  [5] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        </span></span>
<span><span class="co">##  [7] IRanges_2.34.1              S4Vectors_0.38.1           </span></span>
<span><span class="co">##  [9] BiocGenerics_0.46.0         MatrixGenerics_1.12.3      </span></span>
<span><span class="co">## [11] matrixStats_1.0.0           EBImage_4.42.0             </span></span>
<span><span class="co">## [13] cytoviewer_1.1.1            BiocStyle_2.28.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] bitops_1.0-7              gridExtra_2.3            </span></span>
<span><span class="co">##   [3] rlang_1.1.1               magrittr_2.0.3           </span></span>
<span><span class="co">##   [5] svgPanZoom_0.3.4          shinydashboard_0.7.2     </span></span>
<span><span class="co">##   [7] compiler_4.3.1            DelayedMatrixStats_1.22.1</span></span>
<span><span class="co">##   [9] png_0.1-8                 systemfonts_1.0.4        </span></span>
<span><span class="co">##  [11] fftwtools_0.9-11          vctrs_0.6.3              </span></span>
<span><span class="co">##  [13] stringr_1.5.0             pkgconfig_2.0.3          </span></span>
<span><span class="co">##  [15] SpatialExperiment_1.10.0  crayon_1.5.2             </span></span>
<span><span class="co">##  [17] fastmap_1.1.1             magick_2.7.4             </span></span>
<span><span class="co">##  [19] XVector_0.40.0            ellipsis_0.3.2           </span></span>
<span><span class="co">##  [21] fontawesome_0.5.1         scuttle_1.10.2           </span></span>
<span><span class="co">##  [23] utf8_1.2.3                promises_1.2.0.1         </span></span>
<span><span class="co">##  [25] rmarkdown_2.23            ggbeeswarm_0.7.2         </span></span>
<span><span class="co">##  [27] ragg_1.2.5                purrr_1.0.1              </span></span>
<span><span class="co">##  [29] xfun_0.39                 beachmat_2.16.0          </span></span>
<span><span class="co">##  [31] zlibbioc_1.46.0           cachem_1.0.8             </span></span>
<span><span class="co">##  [33] jsonlite_1.8.7            later_1.3.1              </span></span>
<span><span class="co">##  [35] rhdf5filters_1.12.1       DelayedArray_0.26.7      </span></span>
<span><span class="co">##  [37] Rhdf5lib_1.22.0           BiocParallel_1.34.2      </span></span>
<span><span class="co">##  [39] terra_1.7-39              jpeg_0.1-10              </span></span>
<span><span class="co">##  [41] tiff_0.1-11               parallel_4.3.1           </span></span>
<span><span class="co">##  [43] R6_2.5.1                  bslib_0.5.0              </span></span>
<span><span class="co">##  [45] stringi_1.7.12            RColorBrewer_1.1-3       </span></span>
<span><span class="co">##  [47] limma_3.56.2              jquerylib_0.1.4          </span></span>
<span><span class="co">##  [49] Rcpp_1.0.11               bookdown_0.34            </span></span>
<span><span class="co">##  [51] knitr_1.43                R.utils_2.12.2           </span></span>
<span><span class="co">##  [53] nnls_1.4                  httpuv_1.6.11            </span></span>
<span><span class="co">##  [55] Matrix_1.5-4.1            tidyselect_1.2.0         </span></span>
<span><span class="co">##  [57] viridis_0.6.4             abind_1.4-5              </span></span>
<span><span class="co">##  [59] yaml_2.3.7                codetools_0.2-19         </span></span>
<span><span class="co">##  [61] miniUI_0.1.1.1            lattice_0.21-8           </span></span>
<span><span class="co">##  [63] tibble_3.2.1              shiny_1.7.4.1            </span></span>
<span><span class="co">##  [65] evaluate_0.21             archive_1.1.5            </span></span>
<span><span class="co">##  [67] desc_1.4.2                shinycssloaders_1.0.0    </span></span>
<span><span class="co">##  [69] pillar_1.9.0              BiocManager_1.30.21.1    </span></span>
<span><span class="co">##  [71] generics_0.1.3            sp_2.0-0                 </span></span>
<span><span class="co">##  [73] rprojroot_2.0.3           RCurl_1.98-1.12          </span></span>
<span><span class="co">##  [75] ggplot2_3.4.2             sparseMatrixStats_1.12.2 </span></span>
<span><span class="co">##  [77] munsell_0.5.0             scales_1.2.1             </span></span>
<span><span class="co">##  [79] xtable_1.8-4              glue_1.6.2               </span></span>
<span><span class="co">##  [81] tools_4.3.1               colourpicker_1.2.0       </span></span>
<span><span class="co">##  [83] locfit_1.5-9.8            fs_1.6.3                 </span></span>
<span><span class="co">##  [85] rhdf5_2.44.0              grid_4.3.1               </span></span>
<span><span class="co">##  [87] edgeR_3.42.4              DropletUtils_1.20.0      </span></span>
<span><span class="co">##  [89] colorspace_2.1-0          GenomeInfoDbData_1.2.10  </span></span>
<span><span class="co">##  [91] raster_3.6-23             beeswarm_0.4.0           </span></span>
<span><span class="co">##  [93] HDF5Array_1.28.1          vipor_0.4.5              </span></span>
<span><span class="co">##  [95] cli_3.6.1                 textshaping_0.3.6        </span></span>
<span><span class="co">##  [97] fansi_1.0.4               viridisLite_0.4.2        </span></span>
<span><span class="co">##  [99] S4Arrays_1.0.5            svglite_2.1.1            </span></span>
<span><span class="co">## [101] dplyr_1.1.2               gtable_0.3.3             </span></span>
<span><span class="co">## [103] R.methodsS3_1.8.2         sass_0.4.7               </span></span>
<span><span class="co">## [105] digest_0.6.33             dqrng_0.3.0              </span></span>
<span><span class="co">## [107] rjson_0.2.21              htmlwidgets_1.6.2        </span></span>
<span><span class="co">## [109] R.oo_1.25.0               memoise_2.0.1            </span></span>
<span><span class="co">## [111] htmltools_0.5.5           pkgdown_2.0.7            </span></span>
<span><span class="co">## [113] lifecycle_1.0.3           mime_0.12</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Angelo2014" class="csl-entry">
Angelo, Michael, Sean C. Bendall, Rachel Finck, Matthew B. Hale, Chuck
Hitzman, Alexander D. Borowsky, Richard M. Levenson, et al. 2014.
<span>“Multiplexed Ion Beam Imaging of Human Breast Tumors.”</span>
<em>Nature Medicine</em> 20 (4): 436–42.
</div>
<div id="ref-Damond2019" class="csl-entry">
Damond, Nicolas, Stefanie Engler, Vito R. T. Zanotelli, Denis Schapiro,
Clive H. Wasserfall, Irina Kusmartseva, Harry S. Nick, et al. 2019.
<span>“A Map of Human Type 1 Diabetes Progression by Imaging Mass
Cytometry.”</span> <em>Cell Metabolism</em> 29 (3): 755–768.e5.
</div>
<div id="ref-Giesen2014" class="csl-entry">
Giesen, Charlotte, Hao A. O. Wang, Denis Schapiro, Nevena Zivanovic,
Andrea Jacobs, Bodo Hattendorf, Peter J. Schüffler, et al. 2014.
<span>“Highly Multiplexed Imaging of Tumor Tissues with Subcellular
Resolution by Mass Cytometry.”</span> <em>Nature
<span>M</span>ethods</em> 11 (4): 417–22.
</div>
<div id="ref-Jackson2020" class="csl-entry">
Jackson, Hartland W., Jana R. Fischer, Vito R. T. Zanotelli, H. Raza
Ali, Robert Mechera, Savas D. Soysal, Holger Moch, et al. 2020.
<span>“The Single-Cell Pathology Landscape of Breast Cancer.”</span>
<em>Nature</em> 578 (7796): 615–20.
</div>
<div id="ref-Lin2018" class="csl-entry">
Lin, Jia-Ren, Benjamin Izar, Shu Wang, Clarence Yapp, Shaolin Mei, Parin
M. Shah, Sandro Santagata, and Peter K. Sorger. 2018. <span>“Highly
Multiplexed Immunofluorescence Imaging of Human Tissues and Tumors Using
t-CyCIF and Conventional Optical Microscopes.”</span> <em>eLife</em> 7:
1–46.
</div>
<div id="ref-Moffitt2022" class="csl-entry">
Moffitt, Jeffrey R., Emma Lundberg, and Holger Heyn. 2022. <span>“The
Emerging Landscape of Spatial Profiling Technologies.”</span> <em>Nature
Reviews Genetics</em> 23: 741–59.
</div>
<div id="ref-Windhager2021" class="csl-entry">
Windhager, Jonas, Bernd Bodenmiller, and Nils Eling. 2021. <span>“An
End-to-End Workflow for Multiplexed Image Processing and
Analysis.”</span> <a href="https://doi.org/10.1101/2021.11.12.468357" class="external-link">https://doi.org/10.1101/2021.11.12.468357</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Lasse Meyer, Nils Eling.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
